File: achievement-web/src/api/achievement/patent.ts
import http from '@/api/index'

export interface PatentFormDTO {
  patentName: string
  inventors: string
  applicationNo: string
  authorizationNo?: string
  applicationDate: string
  authorizationDate?: string
  patentType: string
  country: string
  nextFeeDate?: string
  legalStatus: string
  isClassified?: number
  classifiedLevel?: string
  projectRef?: string
}

export interface PatentVO {
  id: number
  patentName: string
  inventors: string
  applicationNo: string
  authorizationNo?: string
  applicationDate: string
  authorizationDate?: string
  patentType: string
  country: string
  nextFeeDate?: string
  legalStatus: string
  isClassified?: number
  classifiedLevel?: string
  projectRef?: string
  status: string
  statusLabel: string
  archiveNo?: string
  deptId?: number
  createdBy?: number
  createdTime?: string
  updatedBy?: number
  updatedTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

/**
 * Submit patent for approval (DRAFT -> PENDING_DEPT_REVIEW)
 */
export function submit(data: PatentFormDTO) {
  return http.post('/api/patents/submit', data)
}

/**
 * Create patent as draft
 */
export function createDraft(data: PatentFormDTO) {
  return http.post('/api/patents', data)
}

/**
 * Update an existing patent
 */
export function updateDraft(id: number, data: PatentFormDTO) {
  return http.put(`/api/patents/${id}`, data)
}

/**
 * Save current form as draft
 */
export function saveDraft(data: PatentFormDTO) {
  return http.post('/api/patents/draft', data)
}

/**
 * Get a patent by ID
 */
export function getById(id: number) {
  return http.get(`/api/patents/${id}`)
}

/**
 * Paginated patent listing with filters
 */
export function getPage(params: {
  page: number
  size: number
  status?: string
  keyword?: string
}) {
  return http.get('/api/patents/page', { params })
}

File: achievement-module/achievement-patent/src/main/java/com/institute/achievement/patent/entity/Patent.java
package com.institute.achievement.patent.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Patent achievement entity — maps to the {@code patent} table.
 * <p>
 * Contains all REG-02 metadata fields plus common lifecycle, audit,
 * and classification fields shared across all achievement types.
 */
@Data
@TableName("patent")
public class Patent {

    @TableId(type = IdType.AUTO)
    private Long id;

    // ── Core Metadata (REG-02) ──────────────────────────────────────

    private String patentName;
    private String inventors;
    private String applicationNo;
    private String authorizationNo;
    private LocalDate applicationDate;
    private LocalDate authorizationDate;
    private String patentType;
    private String country;
    private LocalDate nextFeeDate;
    private String legalStatus;

    // ── Classification (D-06) ───────────────────────────────────────

    private Integer isClassified;
    private String classifiedLevel;

    // ── Project Linkage (D-08) ──────────────────────────────────────

    private String projectRef;

    // ── Lifecycle ───────────────────────────────────────────────────

    private String status;
    private String archiveNo;

    // ── Department & Ownership ──────────────────────────────────────

    private Long deptId;
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdTime;

    private Long updatedBy;

    @TableField(fill = FieldFill.UPDATE)
    private LocalDateTime updatedTime;

    // ── Optimistic Locking ──────────────────────────────────────────

    @Version
    @TableField(fill = FieldFill.INSERT)
    private Integer version;
}
